frontend/src/state/documents.ts

```diff
@@ -34,6 +34,41 @@ export function createDocumentsState(editor: EditorState) {
 		download(triggerFileDownload.name, triggerFileDownload.document);
 	});
 
+	editor.dispatcher.subscribeJsMessage(TriggerRasterDownload, (triggerRasterDownload) => {
+		// A canvas to render our svg to in order to get a raster image
+		// https://stackoverflow.com/questions/3975499/convert-svg-to-image-jpeg-png-etc-in-the-browser
+		const canvas = document.createElement("canvas");
+		canvas.width = triggerRasterDownload.size.x;
+		canvas.height = triggerRasterDownload.size.y;
+		const ctx = canvas.getContext("2d");
+		if (!ctx) return;
+
+		// Fill the canvas with white if jpeg (does not support transparency and defaults to black)
+		if (triggerRasterDownload.mime.endsWith("jpeg")) {
+			ctx.fillStyle = "rgb(255,255,255)";
+			ctx.fillRect(0, 0, triggerRasterDownload.size.x, triggerRasterDownload.size.y);
+		}
+
+		// Create a blob url for our svg
+		const img = new Image();
+		const svgBlob = new Blob([triggerRasterDownload.document], { type: "image/svg+xml;charset=utf-8" });
+		const url = URL.createObjectURL(svgBlob);
+		img.onload = (): void => {
+			// Draw our svg to the canvas
+			ctx?.drawImage(img, 0, 0, triggerRasterDownload.size.x, triggerRasterDownload.size.y);
+
+			// Convert the canvas to an image of the correct mime
+			const imgURI = canvas.toDataURL(triggerRasterDownload.mime);
+			// Download our canvas
+			downloadBlob(imgURI, triggerRasterDownload.name);
+
+			// Cleanup resources
+			URL.revokeObjectURL(url);
+			document.removeChild(canvas);
```

Comment:
Since `canvas` was never attached to the DOM, there's no need to remove it from the DOM.

Reply:
It doesn't actually fail silently, an error is logged in the console.

Reply:
Oh interesting! All the more reason to remove it then 😉